design/kiota-e2e.md

```diff
@@ -0,0 +1,147 @@
+# End-to-end usage of kiota-generated SDKs in Javascript
+
+While working with the Kiota + Javascript teams, it became clear that we needed more context and provide a better vision on how we should be using a kiota-generated SDK. This design document aims at providing clarity and a better understanding on how developers will be using our SDKs and the underlying kiota packages.
+
+## Constraints
+
+Before we jump into the end-to-end walk-through, it's important to set some constraints.
+
+| Type      | Description                                                                                                                                   |
+| --------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| Platforms | Node : Current and Previous LTS (14 / 16)<br /> Web : Edge, Chrome, Firefox and Safari (Latest released version + immediate previous version) |
+| Modules   | Node : CommonJS<br /> Web : ES Module (ES6)                                                                                                   |
+| Types     | Typings should be available for both the core and the service libraries for Graph models                                                      |
+
+## NodeJS e2e using the Service library
+
+```bash
+npm install @microsoft/msgraph-sdk-typescript --save
+npm install @microsoft/kiota-authentication-azure --save
+```
+
+```typescript
+// App.ts
+
+import { Client, User, Message, BodyType } from "@microsoft/msgraph-sdk-javascript";
+import { AzureIdentityAuthenticationProvider } from "@microsoft/kiota-authentication-azure";
+import { DeviceCodeCredential } from "@azure/identity";
+
+const deviceCodeCredentials = new DeviceCodeCredential({
+	tenantId: "b61f9af1-d6cf-4cc0-a6f6-befb38bc00ed",
+	clientId: "bde251a6-0ef9-42a8-a40b-9ad9bb594b2c",
+});
+
+const scopes = ["User.Read", "Mail.Send"];
+
+const graphClient = Client.init({
```

Comment:
The user should also be given a choice to  pass serialization implementations

Reply:
Yes, absolutely. I was trying to go the "main" path here. I don't see a lot of developers providing their own serialization implementations, to be honest. This should be part of the Client Options you can pass it in, though. Same for fetch.

Do you feel we should be going in depth in this scenario of the current `init` is enough for the purpose of the e2e?

Reply:
Current `init` is in the core. 

We will need similar factory method for the  `GraphServiceClient` initialization.